### PR TITLE
feat: Add global flag for tracking runtime performance in a metric

### PR DESCRIPTION
**Pull request by jpinkney-aws**
## Problem
We should get some runtime information of q telemetry so that we can figure out what's going wrong through metrics, instead of customer reports

## Solution
I've added an optional flag that when enabled tells toolkits that they should start tracking performance information

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/telemetryformat.md [hunk 1]
```diff
@@ -89,6 +89,8 @@ result: Result
 passive: boolean
 // Value based on unit and call type
 value: number
+// Indicates that the metric should track run-time performance information
+trackPerformance: boolean
```
Comment (awschristou) [inline telemetry/telemetryformat.md:93]: is it `performanceTracking` or `trackPerformance` ?

**Review by awschristou — COMMENTED** [review]
I'd like this change to include updates to the C# and Java generators if possible. Would you like some help with that?

**Review by awschristou — APPROVED** [review]
Thanks for updating the generators!
Things look good to me for C#, please seek approval from @rli for java and @justinmk3 for typescript

File: telemetry/telemetrySchema.json [hunk 2]
```diff
@@ -65,6 +65,10 @@
                     "passive": {
                         "type": "boolean",
                         "description": "Set as true if the metric is emitted without any explicit user action."
+                    },
+                    "trackPerformance": {
+                        "type": "boolean",
+                        "description": "Set as true if the metric can track run-time performance information."
```
Comment (justinmk3) [inline telemetry/telemetrySchema.json:71]: should we mention "defaults to false" ?
```suggestion
                        "description": "Set as true if the metric can track run-time performance information. Default=false"
```

**Review by justinmk3 — APPROVED** [review]

**Comment by rli** [comment]
please exclude jetbrains for now as the current design does not allow us to read the flag until it is too late

**Comment by jpinkney-aws** [comment]
I've just dropped the jetbrains commit and added default=false to the telemetrySchema. Everything else stayed the same

**Pull request merged**